src/common/helpers/snap-builds.js

```diff
@@ -0,0 +1,91 @@
+// Snap Build data
+//
+// Parsed from API response in format:
+// https://launchpad.net/+apidoc/devel.html#snap_build
+// {
+//   buildId: '1232', // last part of 'self_link'
+//
+//   buildLogUrl: '', // 'build_log_url'
+//
+//   // TODO: commit info (once available)
+//   // username: 'John Doe',
+//   // commitId:  'f1d6edb',
+//   // commitMessage:  'Failed commit',
+//
+//   // TODO: use 'distro_arch_series.architecturetag' once available
+//   architecture: 'i386', // last part of 'distro_arch_series_link'
+//
+//   // TODO: full list of statuses based on
+//   // https://git.launchpad.net/launchpad/tree/lib/lp/buildmaster/enums.py#n22
+//   status:  'error', // parsed based on 'buildstate' -> success, error, pending
+//   statusMessage: 'Failed to build', // 'buildstate'
+//
+//   dateCreated: '2016-12-01T17:08:36.317805+00:00', // 'datecreated'
+//   dateStarted: '2016-12-01T17:08:36.317805+00:00', // 'date_started'
+//   dateFirstDispatched: '2016-12-01T17:08:36.317805+00:00', // 'date_first_dispatched'
+//   dateCompleted: '2016-12-01T17:10:36.317805+00:00', // 'datebuilt'
+//   duration: '0:02:00.124039' // 'duration'
+// };
+
+// Based on BuildStatus from LP API
+// https://git.launchpad.net/launchpad/tree/lib/lp/buildmaster/enums.py#n22
+export const BuildStatusLP = {
+  NEEDSBUILD: 'Needs building',     // pending
+  FULLYBUILT: 'Successfully built', // success
+  FAILEDTOBUILD:'Failed to build',  // error
+  MANUALDEPWAIT: 'Dependency wait', // pending
+  CHROOTWAIT: 'Chroot problem',     // error
+  SUPERSEDED: 'Build for superseded Source', // error
+  BUILDING: 'Currently building',   // pending
+  FAILEDTOUPLOAD: 'Failed to upload', // error
+  UPLOADING: 'Uploading build',     // pending
+  CANCELLING: 'Cancelling build',   // pending
+  CANCELLED: 'Cancelled build'      // error
+};
+
+export const BuildStatus = {
+  SUCCESS: 'success', // for builds successfully finished
+  PENDING: 'pending', // for build currently running or in any way in progress
+  ERROR: 'error'      // for builds failed for any reason
+};
+
+// TODO:
+// export enum with build status success/pending/error
+// or export full list of statuses (like in API) and let view do the mapping?
+
+// mapping between build status from LP and pending/success/error internal status
+const BuildStatusMapping = {
```

Comment:
My idea behind having `BuildStatusLP` was to have possibility of using `BuildStatusLP.NEEDSBUILD` instead of hardcoding `"Needs building"` string around the code. So the `BuildStatusLP` would be equivalent of `BuildStatus` enum from Python code to be used around the code with the string values only in one place.

But you are right, that at this point we don't use it anywhere else and I can't think of the case where we may need it if we map it to internal `PENDING/SUCCESS/ERROR` status here, so I can simplify the mapping and drop it.

@cjwatson the `[]` is needed to compute the keys: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names (it's a new JS syntax).